src/finiteVolume/processes/les.cpp

```diff
@@ -0,0 +1,240 @@
+#include "les.hpp"
+#include "finiteVolume/compressibleFlowFields.hpp"
+#include "utilities/mathUtilities.hpp"
+#include "utilities/petscError.hpp"
+
+ablate::finiteVolume::processes::LES::LES(std::shared_ptr<eos::EOS> eosIn)
+
+    : eos(std::move(eosIn)) {}
+
+void ablate::finiteVolume::processes::LES::Setup(ablate::finiteVolume::FiniteVolumeSolver& flow) {
+    if (flow.GetSubDomain().ContainsField(CompressibleFlowFields::EV_FIELD)) {
```

Comment:
I would add a check for the the component of the EV_FIELD here.  If not found I would throw an invalid argument exception.